Frameworks/Foundation/NSKVOSupport.mm

```diff
@@ -424,7 +450,7 @@ - (void*)observationInfo {
 @Status Interoperable
 */
 - (void)setObservationInfo:(void*)observationInfo {
-    objc_setAssociatedObject(self, &s_kvoObservationInfoAssociationKey, (__bridge id)observationInfo, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+    objc_setAssociatedObject(self, &s_kvoObservationInfoAssociationKey, (__bridge id)observationInfo, OBJC_ASSOCIATION_RETAIN);
```

Comment:
Changed to this from an @ synchronized elsewhere. More of a consistency change (this had maybe ~3% perf impact) - there was another call to this function that didn't do the synchronized for whatever reason, and the synchronization should probably occur within the body of this function rather than around it. If you want, I can change this to @ synchronized instead of using the associated objects table's locking, but I don't think it makes much of a difference.